plugin/builtin/sortordertransformer/SortOrderTransformer.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate pluginator
+package main
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+	"sigs.k8s.io/yaml"
+)
+
+// Sort the resources using a customizable ordering based of Kind.
+// Defaults to the ordering of the GVK struct, which puts cluster-wide basic
+// resources with no dependencies (like Namespace, StorageClass, etc.) first,
+// and resources with a high number of dependencies
+// (like ValidatingWebhookConfiguration) last.
+type plugin struct {
+	SortOptions *types.SortOptions `json:"sortOptions,omitempty" yaml:"sortOptions,omitempty"`
+}
+
+//noinspection GoUnusedGlobalVariable
+var KustomizePlugin plugin
+
+// Nothing needed for configuration.
+func (p *plugin) Config(
+	_ *resmap.PluginHelpers, c []byte) (err error) {
+	err = yaml.Unmarshal(c, p)
+	if err != nil {
+		return
+	}
+
+	// Default to FIFO sort, aka no-op.
+	if p.SortOptions == nil {
+		p.SortOptions = &types.SortOptions{
+			Order: types.FIFOSortOrder,
+		}
+	}
+
+	// Validate that the only options set are the ones corresponding to the
+	// selected sort order.
+	if p.SortOptions.Order == types.FIFOSortOrder &&
+		p.SortOptions.LegacySortOptions != nil {
+		return errors.Errorf("the field 'sortOptions.legacySortOptions' but the"+
+			" selected sort order is '%v', not 'legacy'", p.SortOptions.Order)
+	}
+
+	// If legacy sort is selected and no options are given, default to
+	// hardcoded order.
+	if p.SortOptions.Order == types.LegacySortOrder && p.SortOptions.LegacySortOptions == nil {
+		p.SortOptions.LegacySortOptions = &types.LegacySortOptions{
+			OrderFirst: resid.OrderFirst,
+			OrderLast:  resid.OrderLast,
+		}
+	}
+
+	return
+}
+
+func (p *plugin) Transform(m resmap.ResMap) (err error) {
+	// Sort
+	resources := m.Resources()
+	if p.SortOptions.Order == types.LegacySortOrder {
+		s := &idSorter{resources: resources, sortOptions: p.SortOptions}
+		sort.Sort(s)
+	}
+
+	// Clear and refill with the correct order
+	m.Clear()
+	for _, r := range resources {
+		m.Append(r)
+	}
```

Comment:
We don't need to clear and refill with the correct order in the fifo case right? So can we put lines 73-77 in the above if statement to save us some unnecessary work?